### ISSUE TITLE
GraphQL-code-generator should treat type with mapped field as mapped field itself

### ISSUE THREAD

**Issue by SirensOfTitan** [dotansimha/graphql-code-generator#1770]
Updated repro case here: https://github.com/SirensOfTitan/gql-codegen-union-mapper

I'm unsure if this is a bug or whether I should be expecting this.  I know I'd like this of course.
So consider my bug report about mapped union types the other day, with the schema:
```graphql
union MovieLike = Movie | Book
type NonInterfaceHasNarrative {
  narrative: MovieLike!
  movie: Movie!
}
/// *we have a mapper from Movie to MovieEntity*
```

The result looks like this in current latest version:
```
export type ResolversTypes = {
  String: Scalars["String"];
  Boolean: Scalars["Boolean"];
  NonInterfaceHasNarrative: Omit<NonInterfaceHasNarrative, "movie"> & {
    movie: ResolversTypes["Movie"];
  };
  MovieLike: ResolversTypes["Movie"] | ResolversTypes["Book"];
  Movie: MovieEntity;
  ID: Scalars["ID"];
  Int: Scalars["Int"];
  Book: Book;
};
```

... in this case the `movie` field in NonInterfaceHasNarrative is properly replaced with its resolver types.   MovieLike, is not, however.  This leads to the ResolversType for MovieLike in NonInterfaceHasNarrative to be incorrect (`narrative` inside NonInterfaceHasNarrative) as `narrative` should be implicitly mapped to `ResolversTypes['MovieLike']`.  

The idea here is that MovieLike in this instance actually has an implicit map (that is, not defined by the user explicitly but mapped because it has a field that has mapped) of: `ResolversTypes["Movie"] | ResolversTypes["Book"]` that seems like it should be taken into account.

Please let me know if I can provide more detail, I'm just mapping out this bug so might be a poor communicator here. 👅   

Thank you for all your hard work on this library!

**Comment by dotansimha** [comment]
Hi @SirensOfTitan , sorry for the delay, fixed it in: https://github.com/dotansimha/graphql-code-generator/pull/1891

**Comment by dotansimha** [comment]
Fixed in `1.2.0`.
